climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,1059 @@
+        'FIX': Replicate the value onto all the new points. Default is 'AVG'.
+        Works in combination with the kwarg 'disagg_val'.
+    disagg_val: float, optional
+        Specifies what number should be taken as the value, which
+        is to be disaggregated according to the method provided in disagg_met.
+        None: The shape's value is taken from the exp.gdf.value column.
+        float: This given number will be disaggregated according to the method.
+        The default is None.
+
+    Returns
+    -------
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first level indicating
+        membership of the original geometries of exp,
+        second for the point disaggregation within each geometries.
+
+    """
+
+    gdf_geom = exp.gdf.copy()
+
+    if disagg_val is not None:
+        gdf_geom['value'] = disagg_val
+
+    if ((disagg_val is None) and ('value' not in gdf_geom.columns)):
+        raise ValueError('There is no value column in the exposure gdf to'+
+                         ' disaggregate from. Please set disagg_val explicitly.')
+
+    gdf_pnt = gdf_to_pnts(gdf_geom, res, to_meters)
+
+    # disaggregate value column
+    if DisaggMethod(disagg_met) is DisaggMethod.DIV:
+        gdf_pnt = _disagg_values_div(gdf_pnt)
+
+    # set lat lon and centroids
+    exp_pnt = exp.copy()
+    exp_pnt.set_gdf(gdf_pnt)
+    exp_pnt.set_lat_lon()
+
+    return exp_pnt
+
+
+def exp_geom_to_grid(exp, grid, disagg_met, disagg_val):
+    """
+    Disaggregate exposures with (multi-)polygons and/or (multi-)lines
+    geometries to points.
+
+    Parameters
+    ----------
+    exp : Exposures
+        The exposure instance with exp.gdf.geometry containing lines or polygons
+    grid : np.array()
+        Grid on which to disaggregate the exposures. Provided as two
+        vectors [x_grid, y_grid].
+    disagg_met : DisaggMethod
+        Disaggregation method of the shapes's original value onto its inter-
+        polated points. 'DIV': Divide the value evenly over all the new points;
+        'FIX': Replicate the value onto all the new points. Default is 'AVG'.
+        Works in combination with the kwarg 'disagg_val'.
+    disagg_val: float, optional
+        Specifies what number should be taken as the value, which
+        is to be disaggregated according to the method provided in disagg_met.
+        None: The shape's value is taken from the exp.gdf.value column.
+        float: This given number will be disaggregated according to the method.
+        The default is None.
+
+    Returns
+    -------
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first level indicating
+        membership of the original geometries of exp,
+        second for the point disaggregation within each geometries.
+
+    """
+
+    gdf_geom = exp.gdf.copy()
+
+    if disagg_val is not None:
+        gdf_geom.value = disagg_val
```

Comment:
Is the copy still necessary if `disagg_val` _is_ `None`?

Reply:
strictly speaking not, but how would you handle the next line then?
https://github.com/CLIMADA-project/climada_python/blob/3420912b8544897b328faee83a1df840e5604038/climada/util/lines_polys_handler.py#L451 

we would have to run the `gdf_to_grid` then either with `gdf_geom` or with `exp.gdf` depending on the case. That's also adding an "unnecessary" line?

Reply:
One could
```python 
    if disagg_val is not None:
        exp = exp.copy()
        exp.gdf.value = disagg_val
```
and then
```
    gdf_pnt = gdf_to_grid(exp.gdf, grid)
```
This could save quite some memory and time in case disagg_val is None and cost only a little more in case it isn't. (I don't care all too much about whether the line is necessary, it's the calculation it triggers that worries me 😁  )

Reply:
Fair enough, thanks!